realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -1490,21 +1495,22 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeIsNull(JNIEnv* en
                                                                       jlongArray columnIndexes,
                                                                       jlongArray tablePointers)
 {
-    JniLongArray table_arr(env, tablePointers);
-    JniLongArray index_arr(env, columnIndexes);
-    jsize arr_len = index_arr.len();
-    Query* pQuery = Q(nativeQueryPtr);
     try {
-        TableRef src_table_ref = getTableForLinkQuery(nativeQueryPtr, table_arr, index_arr);
+        JniLongArray table_arr(env, tablePointers);
+        JniLongArray index_arr(env, columnIndexes);
+        jsize arr_len = index_arr.len();
+        Query* pQuery = Q(nativeQueryPtr);
+
         jlong column_idx = index_arr[arr_len - 1];
-        TableRef table_ref = getTableByArray(nativeQueryPtr, index_arr);
 
+        TableRef table_ref = getTableByArray(nativeQueryPtr, table_arr, index_arr);
         if (!isNullable(env, TBL(table_arr[arr_len - 1]), table_ref, column_idx)) {
             return;
         }
 
-        // FIXME!!!  Support a backlink as the last column in a field descriptor
-        int col_type = table_ref->get_column_type(S(column_idx));
+        // FIXME: Support a backlink as the last column in a field descriptor
```

Comment:
I assume the behaviour would either to throw an error or just assume this will never match.